pkg/kubelet/kubelet.go

```diff
@@ -2389,7 +2395,7 @@ func (kl *Kubelet) HandlePodDeletions(pods []*api.Pod) {
 		}
 		// Deletion is allowed to fail because the periodic cleanup routine
 		// will trigger deletion again.
-		if err := kl.deletePod(pod.UID); err != nil {
+		if err := kl.deletePod(pod); err != nil {
```

Comment:
@yujuhong @smarterclayton I changed the HandlePodDeletions to pass the entire `api#Pod` to the `deletePod`, so that later the deletion routine will have the pod spec, rather than having to read the container label. This change makes the HandlePodDeletions respect the grace period set through DeleteOptions. 
HandlePodCleanup still doesn't respect the grace period because it doesn't have the pod spec. I think this could be fixed in another PR, possibly by the node team.